typescript/sdk/src/token/IToken.ts

```diff
@@ -82,7 +82,6 @@ export interface IToken extends TokenArgs {
   addConnection(connection: TokenConnection): IToken;
   removeConnection(token: IToken): IToken;
 
-  equals(token: IToken): boolean;
-
-  collateralizes(token: IToken): boolean;
+  equals(token?: IToken): boolean;
+  equalsAsset(token?: IToken): boolean;
```

Comment:
I personally dont think this makes things more clear
hard to tell what `equals` vs `equalsAsset` does
also why optional now?

Reply:
`equalsAsset` sounds like it should be `token.asset.equals(other.asset)`

Reply:
It's optional because it's convenient in a few places to compare a token to a value that may be undefined (which of course returns false)

Reply:
> equalsAsset sounds like it should be token.asset.equals(other.asset)
Yeah it's an interesting idea. I'm open to it. Want to talk more about it

Reply:
Renamed to `isFungibleWith` and added more details to the comment